### ISSUE TITLE
Feature request: allow placeholders in the type of functions

### ISSUE THREAD

**Issue by jad-hamza** [mattam82/Coq-Equations#66]
Hi,

I'm trying to switch from Program Fixpoint to Equations, and so far I'm happy with the rewrite capabilities provided by Equations for unfolding recursive calls.

On the other hand, Program Fixpoint allowed leaving holes in the types of the functions themselves, while Equations doesn't. Are there any plans to add this feature?

```coq
Require Equations.Equations.

Definition f (l l': list nat) (H: 0 = 0): Prop. Admitted.

Equations id (l: list nat) (H: f l l _): { r: list nat | f l r _ } :=
  id l H := exist _ nil _.
```
> Cannot infer this placeholder of type [...]

My workaround for now is to define the type of Equations beforehand, using the usual Program Definition. Is there a better way?
```coq
Require Equations.Equations.

Definition f (l l': list nat) (H: 0 = 0): Prop. Admitted.

Program Definition id_arg2 l: Type := f l l _.
Program Definition id_return_type (l: list nat) (H: id_arg2 l): Type := { r: list nat | f l r _ }.

Equations id (l: list nat) (H: id_arg2 l): id_return_type l H :=
  id l H := exist _ nil _.
```

Another question I have is whether Equations can also support automatic translations (through the generation of additional Obligations) between types and their refine counter-parts (to avoid having to write `exist _ nil _`), as Program does?

**Comment by mattam82** [comment]
The first issue might now be possible to solve in the newest version. For the second, yes, I think if you just `Set Program Modeˋ you will get the subtyping algorithm used by Program, for all following definitions.

**Comment by jad-hamza** [comment]
Thanks! I'll try the new version when it's available

**Comment by mattam82** [comment]
This is fixed now in master (upcoming release for 8.8/8.9 will have it)
